e3sm_diags/driver/utils/regrid.py

```diff
@@ -0,0 +1,635 @@
+        in mb units. This parameter is usually set by ``CoreParameter.plevs``
+        attribute. For example, ``plevs=[850.0, 200.0]``.
+
+    Returns
+    -------
+    xr.Dataset
+        The dataset with the variables's Z axis regridded to the desired
+        pressure levels (mb units).
+
+    Raises
+    ------
+    KeyError
+        If the Z axis has no "long_name" attribute to determine whether it is
+        hybrid or pressure.
+    ValueError
+        If the Z axis "long_name" attribute is not "hybrid", "isobaric",
+        or "pressure".
+
+    Notes
+    -----
+    Replaces `e3sm_diags.driver.utils.general.convert_to_pressure_levels`.
+    """
+    ds = dataset.copy()
+    dv = ds[var_key]
+
+    z_axis = get_z_axis(dv)
+    z_long_name = z_axis.attrs.get("long_name")
+
+    if z_long_name is None:
+        raise KeyError(
+            f"The vertical level ({z_axis.name}) for '{dv.name}' does "
+            "not have a 'long_name' attribute to determine whether it is hybrid "
+            "or pressure."
+        )
+
+    z_long_name = z_long_name.lower()
+
+    # Hybrid must be the first conditional statement because the long_name attr
+    # can be "hybrid sigma pressure coordinate" which includes "hybrid" and
+    # "pressure".
+    if "hybrid" in z_long_name:
+        ds_plevs = _hybrid_to_plevs(ds, var_key, plevs)
+    elif "pressure" in z_long_name or "isobaric" in z_long_name:
+        ds_plevs = _pressure_to_plevs(ds, var_key, plevs)
+    else:
+        raise ValueError(
+            f"The vertical level ({z_axis.name}) for '{dv.name}' is "
+            "not hybrid or pressure. Its long name must either include 'hybrid', "
+            "'pressure', or 'isobaric'."
+        )
+
+    return ds_plevs
+
+
+def _hybrid_to_plevs(
+    dataset: xr.Dataset,
+    var_key: str,
+    plevs: List[int] | List[float],
+) -> xr.DataArray:
+    """Regrid a variable's hybrid-sigma levels to the desired pressure levels.
+
+    Steps:
+        1. Create the output pressure grid using ``plevs``.
+        2. Convert hybrid-sigma levels to pressure coordinates.
+        3. Regrid the pressure coordinates to the output pressure grid (plevs).
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        The dataset with the variable using hybrid-sigma levels.
+    var_key : var_key.
+        The variable key.
+    plevs : List[int] | List[float]
+        A 1-D array of floats or integers representing output pressure levels
+        in mb units. For example, ``plevs=[850.0, 200.0]``. This parameter is
+        usually set by the ``CoreParameter.plevs`` attribute.
+
+    Returns
+    -------
+    xr.Dataset
+        The variable with a Z axis regridded to pressure levels (mb units).
+
+    Notes
+    -----
+    Replaces `e3sm_diags.driver.utils.general.hybrid_to_plevs`.
+    """
+    # TODO: Do we need to convert the Z axis to mb units if it is in PA? Or
+    # do we always expect units to be in mb?
```

Comment:
This looks like it is still `TODO`?

Reply:
This was a question for @chengzhuzhang. The old logic assumes Z axis is always in mb units, but I wanted to make sure if this is correct.

Reply:
Yes the assumption is that Z axis, and in put units will all in units mb. Let's keep this TODO for now...